src/Fixer/Whitespace/ReferenceSpacesFixer.php

```diff
@@ -0,0 +1,268 @@
+     * @internal
+     */
+    public const TYPE_ANONYMOUS_FUNCTION_USE_BLOCK = 'anonymous_function_use_block';
+
+    /**
+     * @internal
+     */
+    public const BY_ASSIGN = 'by_assign';
+
+    /**
+     * @internal
+     */
+    public const BY_REFERENCE = 'by_reference';
+
+    /**
+     * @internal
+     */
+    public const SINGLE_SPACE = 'single_space';
+
+    /**
+     * @internal
+     */
+    public const NO_SPACE = 'no_space';
+
+    /** @var array<'&'|287|288> */
+    private array $relevantTokens = [];
+
+    public function configure(array $configuration): void
+    {
+        parent::configure($configuration);
+
+        $this->relevantTokens = \PHP_VERSION_ID >= 80100
+            ? [T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG, T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG, '&']
+            : ['&'];
+    }
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Reference operator should be surrounded by space as defined.',
+            [
+                new CodeSample(
+                    '<?php $foo = & $bar;
+'
+                ),
+                new CodeSample(
+                    '<?php $foo = & $bar;
+',
+                    ['assignment' => self::BY_ASSIGN]
+                ),
+                new CodeSample(
+                    '<?php $foo = & $bar;
+',
+                    ['assignment' => self::BY_REFERENCE]
+                ),
+                new CodeSample(
+                    '<?php $foo =&$bar;
+',
+                    ['assignment' => self::SINGLE_SPACE]
+                ),
+                new CodeSample(
+                    '<?php $foo =& $bar;
+',
+                    ['assignment' => self::NO_SPACE]
+                ),
+                new CodeSample(
+                    '<?php function foo(& $bar) {}
+',
+                    ['function_signature' => self::BY_REFERENCE]
+                ),
+                new CodeSample(
+                    '<?php function foo(&$bar) {}
+',
+                    ['function_signature' => self::SINGLE_SPACE]
+                ),
+                new CodeSample(
+                    '<?php $foo = function () use (& $bar) {};
+',
+                    ['anonymous_function_use_block' => self::BY_REFERENCE]
+                ),
+                new CodeSample(
+                    '<?php $foo = function () use (&$bar) {};
+',
+                    ['anonymous_function_use_block' => self::SINGLE_SPACE]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run after BinaryOperatorSpacesFixer
+     */
+    public function getPriority(): int
+    {
+        return -33;
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAnyTokenKindsFound($this->relevantTokens);
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
```

Comment:
for each of the option here - is any formatting of those preferred by PSR/PER ?

if so, is there a community using other possibilities?
if not - please drop those additional options

As PHP CS Fixer, we do not aim to cover each possible variant of style, but only the recommended / most popular ones, as each configuration option increase the maintenance cost and go against the purpose to spread the best recommendations.

Reply:
Both PSR and PER only mention the reference operator in regards to function signature. Nothing in regards to assignment. I would think that we could easily merge the `function_signature` and `anonymous_function_use_block` options into one. 

After doing some searching I have not been able to find anything else than
- assignment: `$foo =& $bar` or `$foo = &$bar`
- signatures: `function foo(&$bar)`

I just added function signatures based on @Wirone comment in #7303 

Would you say I should altogether remove the `function_signature` and `anonymous...` options and just handle assign by reference (either `=& $bar` or `= &$bar`) in this PR?